src/python/pants/backend/codegen/protobuf/tailor.py

```diff
@@ -36,7 +32,7 @@ async def find_putative_targets(
         PutativeTarget.for_target_type(
             ProtobufLibrary,
             path=dirname,
-            name=os.path.basename(dirname),
+            name="protos",
```

Comment:
I don't feel `protos` follow the same pattern as the default name for other targets.
`python_library` -> `lib`
`python_tests` -> `tests`
`protobuf_library` -> `protos` (why the plural?)
etc..
Simply dropping the s down to `proto` would fit better, imho, to keep it short and succinct.

Reply:
I can see why `lib` was not chosen for protobuf targets..

Reply:
The target will be split into protobuf_sources and protobuf_source soon. I think name=protos would make more sense then. Wdyt?

Reply:
`protos` for `protobuf_sources` works. Was it a typo, or did you also mean `protobuf_source` ? Why two targets so closely named?

Reply:
Intentional. This is what the proposal at https://docs.google.com/document/d/1HpJn2jTWf5sKob6zhe4SqHZ7KWBlX4a6OJOwnuw-IHo/edit is about. protobuf_source (singular) is what actually gets used in builds, because protobufs are modeled at the file level - like imports happen between files etc. protobuf_sources (plural) is a generator of protobuf_source targets, and only exists to reduce boilerplate.

We expect protobuf_sources to be mostly what people use in BUILD files. In fact, we maybe would make protobuf_source be private by using a _ in front. Not sure

(Also I'm open to ideas for better naming)

Reply:
Ah, right, ok makes sense. 

Quick, perhaps dumb, question, Does all the different file targets have to be specific, can't a file be treated as a file in a generic way, and any specific behaviour come from the fact what kind of target owns it?
So that `python_sources`, `protobuf_sources`, `shell_sources` etc all generate `file(name="mod.py", ...)` / `file(name="msg.proto", ...)` etc targets ?

Reply:
Not dumb! Good question. 

The other way around would work, but not that. The target type for the "atomic targets" (`python_source`, `go_package`, `python_requirement`, etc) is what we directly operate on and how we know what semantics to use: such as whether to try running tests on a Python file or not. Note that it's possible for example to have a `.py` file in a `file` target vs `python_source` vs `python_test` - the semantics all change based on which you choose.

It is totally plausible, though, to have one single generic target generator!

```python
gen_targets(
   name="generator",
   metadata={
       ["**/*.py"]: {"target_type": "python_source", "skip_pylint": True},
       ["**/*.java"]: {"target_type": "java_source"}
    }
)
```

Of course, that's not very intuitive not ergonomic, so I think it's better to have `python_sources`, `python_tests`, etc.

Reply:
Thanks for the clarification :)